api/src/main/java/org/jboss/forge/roaster/model/Type.java

```diff
@@ -34,6 +38,8 @@
 
    boolean isQualified();
 
+   boolean isNameQualified();
```

Comment:
isQualified delegates to the implementation, but (from the JDT javadoc) :

	 * Note that a type like "A.B" can be represented either of two ways:
	 * <code>QualifiedType(SimpleType(SimpleName("A")),SimpleName("B"))</code>
	 * <code>SimpleType(QualifiedName(SimpleName("A"),SimpleName("B")))</code>
	 * The first form is preferred when "A" is known to be a type. However, a
	 * parser cannot always determine this. Clients should be prepared to handle
	 * either rather than make assumptions. (Note also that the first form
	 * became possible as of JLS3; only the second form existed in the
	 * JLS2 API.)

That "parser cannot always determine this" happens e.g. with qualified generics, where option B is picked. 
Our use of the method isQualified() always assumes option "A" instead. 
See for example FieldTest.testImportImpliedGenerics

So, "isNameQualified" only looks at the name and defines "qualified" in terms of dot-connected names,
while "isQualified" checks whether, internally, the name is implemented with a JDT QualifiedType or not